pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -74,16 +74,15 @@ func allKubernetesTopologyLabelsOrdered() []string {
 		append([]string{corev1.LabelTopologyRegion,
 			corev1.LabelTopologyZone},
 			rookTopologyLabelsOrdered()...),
-		labelHostname, //  host is the lowest level in the crush map hierarchy
+		k8sutil.LabelHostname(), //  host is the lowest level in the crush map hierarchy
 	)
 }
 
 func kubernetesTopologyLabelToCRUSHLabel(label string) string {
-	crushLabel := strings.Split(label, "/")
-	if crushLabel[len(crushLabel)-1] == "hostname" {
```

Comment:
Will existing osds be immediately converted to the new label? I wonder if there will be some transition issue where the label is still the only one, and we would still need this check on line 83 to ensure a smooth transition. Maybe it is not an issue, but to be safe perhaps consider adding this line 83 back? (in addition to keeping your new check for `k8sutil.LabelHostname()`)

Reply:
I am not sure that i fully inderstand your question. i've tested migration locally (`kubernetes.io/hostname -> custom label` and `custom label1 -> custom label 2` ) and in hapenned in the following way: 
- Rook tries to update osd deployments with new label in `nodeSelector`
- osd restarted not immediately but one-by-one. in the same way like it happens for osd upgrades.